# **TikTok Project** 
## *Preliminary Data Analysis*

Welcome to the TikTok Project

In this fictional scenario, I am assuming the role of a data professional who has just joined a team at TikTok. The team is in the early stages of a new project, which has been approved by TikTok's leadership team. My primary responsibility is to explore and analyze the provided data to prepare for the development of a claims classification model. This involves conducting exploratory data analysis (EDA) to uncover patterns and insights that will guide the next steps in the project.

## **Inspect and Analyze Data**

In this fictional scenario, I have been tasked with examining data provided by TikTok, with the goal of preparing it for further analysis.

**Project Purpose:**

The purpose of this project is to investigate and understand the dataset provided. The key objectives of this activity are:

1. To become familiar with the data.
2. To compile summary information about the dataset.
3. To begin the process of exploratory data analysis (EDA) and uncover insights within the data.
4. To prepare for more in-depth EDA, hypothesis testing, and statistical analysis.

**Goal:**

The goal of this project was to construct a dataframe in Python, perform an initial inspection of the dataset, and communicate the findings to the TikTok data team.

---

### *This activity consists of three parts:*

**Part 1: Understand the Situation**
- How can the provided TikTok data be best organized and understood to facilitate further analysis?

**Part 2: Understand the Data**
- A pandas dataframe was created to facilitate learning and future exploratory data analysis (EDA) and statistical activities.
- Summary information was compiled to inform the next steps of the analysis.

**Part 3: Understand the Variables**
- Insights from the examination of the summary data were used to guide a deeper investigation into the individual variables.

---

To complete this activity, I followed the instructions provided, answered the questions outlined in the project brief, and used those insights to draft an executive summary based on the findings. This project was completed prior to advancing to the next course and was assessed by comparing the results to a completed exemplar.


# **Identify data types and compile summary information**


Throughout these project notebooks, you'll see references to the problem-solving framework PACE. The following notebook components are labeled with the respective PACE stage: Plan, Analyze, Construct, and Execute.

# **PACE stages**

  - Plan
  - Analyze
  - Construct
  - Execute



## **PACE: Plan**

Consider the questions in your PACE Strategy Document and those below to craft your response:



### **Understand the situation**
To understant the situation, the first step is to explore the *Data Dictionary*

**Data Dictionary**

| Column Name              | Type   | Description                                                                                                    |
|--------------------------|--------|----------------------------------------------------------------------------------------------------------------|
| #                        | int    | TikTok assigned number for video with claim/opinion.                                                          |
| claim_status             | obj    | Whether the published video has been identified as an “opinion” or a “claim.” An “opinion” refers to personal beliefs or thoughts, while a “claim” refers to unsourced or unverified information. |
| video_id                 | int    | Random identifying number assigned to the video upon publication on TikTok.                                    |
| video_duration_sec       | int    | How long the published video is measured in seconds.                                                          |
| video_transcription_text | obj    | Transcribed text of the words spoken in the published video.                                                   |
| verified_status          | obj    | Indicates the verification status of the TikTok user who published the video, either “verified” or “not verified.” |
| author_ban_status        | obj    | Indicates the status of the TikTok user who published the video in terms of their permissions: “active,” “under scrutiny,” or “banned.” |
| video_view_count         | float  | The total number of times the published video has been viewed.                                                 |
| video_like_count         | float  | The total number of times the published video has been liked by other users.                                  |
| video_share_count        | float  | The total number of times the published video has been shared by other users.                                 |
| video_download_count     | float  | The total number of times the published video has been downloaded by other users.                             |
| video_comment_count      | float  | The total number of comments on the published video.                                                           |


Afterward, it is important to explore the dataset to identify key variables for the stakeholder.

## **PACE: Analyze**



### Imports and data loading**
- Import the necessary packages to load and explore the dataset, such as numpy and pandas

In [1]:
import pandas as pd
import numpy as np


Then, load the dataset into a dataframe. Creating a dataframe will help you conduct data manipulation, exploratory data analysis (EDA), and statistical activities.



In [2]:
# Load dataset into dataframe
data = pd.read_csv("tiktok_dataset.csv")

### **Understand the Data - Inspect the Data**

To begin inspecting the dataset, summary information about the dataframe can be viewed by using the following functions:

1. `data.head(10)` — Displays the first 10 rows of the dataset.
2. `data.info()` — Provides a summary of the dataframe, including the data types and any null values.
3. `data.describe()` — Shows descriptive statistics for numeric variables, including mean, standard deviation, and quartiles.

*Consider the following questions:*

When reviewing the first few rows of the dataframe, it's important to observe what data is present and what each row represents. This will give an understanding of the structure of the dataset and the types of information it contains.

The `data.info()` output helps identify the types of variables in the dataset, along with any null values. Noticing whether all variables are numeric or contain other data types can highlight potential issues or areas needing further investigation.

The `data.describe()` output reveals the distributions of numeric variables, such as the mean and range of values. By looking at this, one can identify potential outliers or questionable values that may need to be addressed before further analysis.

















In [3]:
data.head(10)

,#,claim_status,video_id,video_duration_sec,video_transcription_text,verified_status,author_ban_status,video_view_count,video_like_count,video_share_count,video_download_count,video_comment_count
0,1,claim,7017666017,59,someone shared with me that drone deliveries a...,not verified,under review,343296.0,19425.0,241.0,1.0,0.0
1,2,claim,4014381136,32,someone shared with me that there are more mic...,not verified,active,140877.0,77355.0,19034.0,1161.0,684.0
2,3,claim,9859838091,31,someone shared with me that american industria...,not verified,active,902185.0,97690.0,2858.0,833.0,329.0
3,4,claim,1866847991,25,someone shared with me that the metro of st. p...,not verified,active,437506.0,239954.0,34812.0,1234.0,584.0
4,5,claim,7105231098,19,someone shared with me that the number of busi...,not verified,active,56167.0,34987.0,4110.0,547.0,152.0
5,6,claim,8972200955,35,someone shared with me that gross domestic pro...,not verified,under review,336647.0,175546.0,62303.0,4293.0,1857.0
6,7,claim,4958886992,16,someone shared with me that elvis presley has ...,not verified,active,750345.0,486192.0,193911.0,8616.0,5446.0
7,8,claim,2270982263,41,someone shared with me that the best selling s...,not verified,active,547532.0,1072.0,50.0,22.0,11.0
8,9,claim,5235769692,50,someone shared with me that about half of the ...,not verified,active,24819.0,10160.0,1050.0,53.0,27.0
9,10,claim,4660861094,45,someone shared with me that it would take a 50...,verified,active,931587.0,171051.0,67739.0,4104.0,2540.0


In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19382 entries, 0 to 19381
Data columns (total 12 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   #                         19382 non-null  int64  
 1   claim_status              19084 non-null  object 
 2   video_id                  19382 non-null  int64  
 3   video_duration_sec        19382 non-null  int64  
 4   video_transcription_text  19084 non-null  object 
 5   verified_status           19382 non-null  object 
 6   author_ban_status         19382 non-null  object 
 7   video_view_count          19084 non-null  float64
 8   video_like_count          19084 non-null  float64
 9   video_share_count         19084 non-null  float64
 10  video_download_count      19084 non-null  float64
 11  video_comment_count       19084 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 1.8+ MB


In [5]:
data.describe()

,#,video_id,video_duration_sec,video_view_count,video_like_count,video_share_count,video_download_count,video_comment_count
count,19382.000000,1.938200e+04,19382.000000,19084.000000,19084.000000,19084.000000,19084.000000,19084.000000
mean,9691.500000,5.627454e+09,32.421732,254708.558688,84304.636030,16735.248323,1049.429627,349.312146
std,5595.245794,2.536440e+09,16.229967,322893.280814,133420.546814,32036.174350,2004.299894,799.638865
min,1.000000,1.234959e+09,5.000000,20.000000,0.000000,0.000000,0.000000,0.000000
25%,4846.250000,3.430417e+09,18.000000,4942.500000,810.750000,115.000000,7.000000,1.000000
50%,9691.500000,5.618664e+09,32.000000,9954.500000,3403.500000,717.000000,46.000000,9.000000
75%,14536.750000,7.843960e+09,47.000000,504327.000000,125020.000000,18222.000000,1156.250000,292.000000
max,19382.000000,9.999873e+09,60.000000,999817.000000,657830.000000,256130.000000,14994.000000,9599.000000


- The TikTok dataframe contains a mix of categorical, text, and numerical data, with each row representing a unique TikTok video. The dataset includes information on whether the video conveys a claim or an opinion, as well as associated metadata.

- The dataframe consists of five `float64` columns, three `int64` columns, and four `object` columns, totaling 19,382 rows (videos). Some variables have missing values, including claim status, video transcription, and all count-related variables.

- Several of the count variables exhibit outliers at the high end of their distributions, with large standard deviations and maximum values significantly higher than their quartile values. Additionally, variables like `video_comment_count` and `video_download_count` have a high concentration of zero values, which may indicate skewness or specific user behavior patterns.

### **Understand the data - Investigate the variables**

In this phase, the focus shifts to a deeper investigation of the variables to gain a better understanding of the dataset.

Since the ultimate goal of the project is to classify videos as either claims or opinions using machine learning, a logical first step is to examine the claim_status variable. By analyzing the distribution of this variable, insights can be gained into the balance between claims and opinions. The analysis begins by determining the number of videos for each unique claim status.

In [6]:
# What are the different values for claim status and how many of each are in the data?
data['claim_status'].value_counts()


claim_status
claim      9608
opinion    9476
Name: count, dtype: int64


The distribution of claim statuses is fairly even.

Next, the engagement trends associated with each claim status are examined.

Boolean masking is used to filter the data based on claim status, followed by the calculation of the mean and median view counts for each claim status.

In [7]:
# What is the average view count of videos with "claim" status?
claims = data[data['claim_status'] == 'claim']
print("mean view count for claims was", claims["video_view_count"].mean())
print("median view count for claims was", claims["video_view_count"].median())


mean view count for claims was 501029.4527477102
median view count for claims was 501555.0


In [8]:
# What is the average view count of videos with "opinion" status?
opinions = data[data['claim_status'] == 'opinion']
print("mean view count for opinions was", opinions["video_view_count"].mean())
print("median view count for opinions was", opinions["video_view_count"].median())

mean view count for opinions was 4956.43224989447
median view count for opinions was 4953.0


The mean and median values for both claims and opinions are similar, indicating that the distribution of these variables is symmetrical. The view count distribution appears to be fairly balanced.

Next, trends associated with the author's ban status are examined.


In [9]:
# Get counts for each group combination of claim status and author ban status


data.groupby(['claim_status', 'author_ban_status']).count()[['#']]
data.groupby(['claim_status', 'author_ban_status'])["video_view_count"].describe()

count           mean            std     min  \
claim_status author_ban_status                                                 
claim        active             6566.0  499221.733171  292588.642995  1207.0   
             banned             1439.0  505907.917304  287372.286655  2557.0   
             under review       1603.0  504054.640674  289907.730293  1049.0   
opinion      active             8817.0    4958.120563    2886.086081    20.0   
             banned              196.0    4876.530612    2853.284313    72.0   
             under review        463.0    4958.105832    2901.908640    75.0   

                                      25%       50%        75%       max  
claim_status author_ban_status                                            
claim        active             242687.25  499500.5  752049.75  999817.0  
             banned             253925.50  512572.0  761580.00  997703.0  
             under review       254470.50  500774.0  749936.00  999655.0  
opinion      active               2462.00    4958.0    7452.00    9998.0  
             banned               2487.00    5083.5    7368.00    9916.0  
             under review         2490.00    4884.0    7435.00    9964.0

There are significantly more claim videos associated with banned authors compared to opinion videos with banned authors. This may suggest that claim videos are more closely monitored, and authors may face stricter rules when posting claims than when posting opinions. However, it is unclear whether claim videos are inherently more likely to result in bans, or if authors of claim videos are more prone to violating terms of service.

Content moderation policies may lead to higher bans for claim videos, as these videos tend to involve controversial or misinformation-prone content, which attracts more scrutiny. This increased attention could lead to a higher likelihood of a ban for authors of claim videos compared to those posting opinions, which are generally less contentious.

It is important to note that conclusions can be drawn about the status of banned versus active authors, but not about the banned videos themselves. It is impossible to determine if a specific video caused the ban, as authors may have posted both compliant and banned videos.

The investigation into engagement levels continues, now focusing on author_ban_status.

The median video share count for each author ban status is calculated.

In [10]:
data.groupby(['author_ban_status']).agg({'video_view_count':['mean', 'median'],
                                        'video_like_count':['mean', 'median'],
                                        'video_share_count':['mean', 'median']})



video_view_count           video_like_count            \
                              mean    median             mean    median   
author_ban_status                                                         
active               215927.039524    8616.0     71036.533836    2222.0   
banned               445845.439144  448201.0    153017.236697  105573.0   
under review         392204.836399  365245.5    128718.050339   71204.5   

                  video_share_count           
                               mean   median  
author_ban_status                             
active                 14111.466164    437.0  
banned                 29998.942508  14468.0  
under review           25774.696999   9444.0

In [11]:
# What's the median video share count of each author ban status?
data.groupby(['author_ban_status']).agg({'video_share_count':['median']})

,video_share_count
,median
author_ban_status,
active,437.0
banned,14468.0
under review,9444.0


Banned authors have a median share count that is 33 times higher than that of active authors. When analyzing the views, likes, and shares, it becomes apparent that videos associated with banned or under review authors experience significantly more interactions than those of active authors. This suggests that videos from banned or under review authors may attract more attention.



In [12]:
data.groupby(['author_ban_status']).median(numeric_only=True)[
['video_share_count']]

,video_share_count
author_ban_status,
active,437.0
banned,14468.0
under review,9444.0


Banned authors and those under review receive significantly more views, likes, and shares compared to active authors. In most groups, the mean is notably higher than the median, indicating the presence of videos with exceptionally high engagement levels.


I created three new columns to better understand engagement rates:

- `likes_per_view`: This column represents the number of likes divided by the number of views for each video.
- `comments_per_view`: This column represents the number of comments divided by the number of views for each video.
- `shares_per_view`: This column represents the number of shares divided by the number of views for each video.

In [13]:
# Create a likes_per_view column
data['likes_per_view'] = data['video_like_count']/ data['video_view_count']


# Create a comments_per_view column
data['comments_per_view'] = data['video_comment_count']/ data['video_view_count']

# Create a shares_per_view column
data['shares_per_view'] = data['video_share_count']/ data['video_view_count']


The data was organized by claim status and author ban status to analyze engagement patterns. Summary statistics, including count, mean, and median, were calculated for the newly created engagement rate columns. This provided insights into how different types of content and author statuses influence audience interactions.

In [14]:
data.groupby(['claim_status', 'author_ban_status']).agg({'likes_per_view': ['count','mean', 'median'],
                                                       'comments_per_view' : ['count','mean', 'median'],
                                                       'shares_per_view': ['count','mean', 'median']})

likes_per_view                      \
                                        count      mean    median   
claim_status author_ban_status                                      
claim        active                      6566  0.329542  0.326538   
             banned                      1439  0.345071  0.358909   
             under review                1603  0.327997  0.320867   
opinion      active                      8817  0.219744  0.218330   
             banned                       196  0.206868  0.198483   
             under review                 463  0.226394  0.228051   

                               comments_per_view                      \
                                           count      mean    median   
claim_status author_ban_status                                         
claim        active                         6566  0.001393  0.000776   
             banned                         1439  0.001377  0.000746   
             under review                   1603  0.001367  0.000789   
opinion      active                         8817  0.000517  0.000252   
             banned                          196  0.000434  0.000193   
             under review                    463  0.000536  0.000293   

                               shares_per_view                      
                                         count      mean    median  
claim_status author_ban_status                                      
claim        active                       6566  0.065456  0.049279  
             banned                       1439  0.067893  0.051606  
             under review                 1603  0.065733  0.049967  
opinion      active                       8817  0.043729  0.032405  
             banned                        196  0.040531  0.030728  
             under review                  463  0.044472  0.035027

Videos created by banned authors and those under review tend to receive significantly more views, likes, and shares than those by active authors. However, engagement rates per view appear to be influenced more by claim status than by author ban status.

Claim videos not only achieve higher view counts than opinion videos but also receive more favorable engagement, with higher average like rates. Additionally, claim videos generate more interactions through comments and shares compared to opinion videos.

For claim videos, banned authors exhibit slightly higher likes-per-view and shares-per-view rates than active authors or those under review. In contrast, for opinion videos, active authors and those under review experience higher engagement rates across all categories compared to banned authors.

## **PACE: Construct**

**Note**: The Construct stage does not apply to this workflow. 

## **PACE: Execute**


### **Summary**

The dataset contains 19,382 samples, with approximately 50% classified as claims (9,608 videos).

Engagement levels show a strong correlation with claim status, indicating that this factor should be a key focus for further analysis.

Videos created by banned authors consistently exhibit significantly higher engagement compared to those by active authors, suggesting a potential relationship between content moderation and audience interaction.
